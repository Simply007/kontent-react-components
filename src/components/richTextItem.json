{
  "item": {
    "system": {
      "id": "e72461f6-49d6-4308-8250-0b6e36cec15d",
      "name": "Ondřej Chrastina",
      "codename": "ondrej_chrastina",
      "language": "default",
      "type": "person",
      "collection": "default",
      "sitemap_locations": [],
      "last_modified": "2022-02-14T09:33:27.4528882Z",
      "workflow_step": "published"
    },
    "elements": {
      "name": {
        "type": "text",
        "name": "Name",
        "value": "Ondřej Chrastina"
      },
      "bio": {
        "type": "rich_text",
        "name": "Bio",
        "images": {
          "d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88": {
            "image_id": "d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88",
            "description": null,
            "url": "https://assets-us-01.kc-usercontent.com:443/0f6785d7-8214-000f-62b3-d9580d692b59/44299668-b37b-4224-a115-1fd66f7d7b36/Yprofile.jpg",
            "width": 500,
            "height": 500
          }
        },
        "links": {
          "2b805947-7ca5-4e6a-baa5-734a91f3cfa2": {
            "codename": "john_doe",
            "type": "person",
            "url_slug": "john-doe"
          }
        },
        "modular_content": [
          "ondrej_chrastina_tech",
          "simply007_kontent_gatsby_benchmark",
          "n5b76e971_4709_01f2_e8ea_36a9716d7b42"
        ],
        "value": "<p>This is Ondřej Chrastina - Developer Advocate with <a href=\"https://kontent.ai\" data-new-window=\"true\" target=\"_blank\" rel=\"noopener noreferrer\">Kentico Kontent</a>.</p>\n<figure data-asset-id=\"d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88\" data-image-id=\"d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88\"><img src=\"https://assets-us-01.kc-usercontent.com:443/0f6785d7-8214-000f-62b3-d9580d692b59/44299668-b37b-4224-a115-1fd66f7d7b36/Yprofile.jpg\" data-asset-id=\"d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88\" data-image-id=\"d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88\" alt=\"\"></figure>\n<p>He likes to do web sites. This is his latest project:</p>\n<object type=\"application/kenticocloud\" data-type=\"item\" data-rel=\"link\" data-codename=\"ondrej_chrastina_tech\"></object>\n<p><br>\nHe also likes OSS. This is his latest repository:</p>\n<object type=\"application/kenticocloud\" data-type=\"item\" data-rel=\"link\" data-codename=\"simply007_kontent_gatsby_benchmark\"></object>\n<p><br>\nOn some projects, he was cooperating with <a data-item-id=\"2b805947-7ca5-4e6a-baa5-734a91f3cfa2\" href=\"\">John Doe</a>.</p>\n<p>You could take a look at their&nbsp;<a href=\"https://google.com\" title=\"sample link\">latest project</a>.</p>\n<p>Check out also this interesting repository</p>\n<object type=\"application/kenticocloud\" data-type=\"item\" data-rel=\"component\" data-codename=\"n5b76e971_4709_01f2_e8ea_36a9716d7b42\"></object>"
      },
      "slug": {
        "type": "url_slug",
        "name": "Slug",
        "value": "ondrej-chrastina"
      }
    }
  },
  "modular_content": {
    "n5b76e971_4709_01f2_e8ea_36a9716d7b42": {
      "system": {
        "id": "5b76e971-4709-01f2-e8ea-36a9716d7b42",
        "name": "5b76e971-4709-01f2-e8ea-36a9716d7b42",
        "codename": "n5b76e971_4709_01f2_e8ea_36a9716d7b42",
        "language": "default",
        "type": "repository",
        "collection": "default",
        "sitemap_locations": [],
        "last_modified": "2022-02-14T09:33:27.4528882Z"
      },
      "elements": {
        "name": {
          "type": "text",
          "name": "Name",
          "value": "Gatsby packages repository"
        },
        "slug": {
          "type": "url_slug",
          "name": "Slug",
          "value": ""
        },
        "summary": {
          "type": "text",
          "name": "Summary",
          "value": "Gatsby packages repository containing the source plugin, components package, and a bunch of Gatsby examples."
        },
        "url": {
          "type": "text",
          "name": "URL",
          "value": "https://github.com/Kentico/kontent-gatsby-packages"
        }
      }
    },
    "ondrej_chrastina_tech": {
      "system": {
        "id": "884bd3cf-98b6-438e-b50d-a593a3bad623",
        "name": "ondrej.chrastina.tech",
        "codename": "ondrej_chrastina_tech",
        "language": "default",
        "type": "website",
        "collection": "default",
        "sitemap_locations": [],
        "last_modified": "2020-08-14T13:18:13.858424Z",
        "workflow_step": "published"
      },
      "elements": {
        "name": {
          "type": "text",
          "name": "Name",
          "value": "ondrej.chrastina.tech"
        },
        "summary": {
          "type": "text",
          "name": "Summary",
          "value": "Ondřej Chrastina's personal site"
        },
        "url": {
          "type": "text",
          "name": "URL",
          "value": "https://ondrej.chrastina.tech"
        },
        "slug": {
          "type": "url_slug",
          "name": "Slug",
          "value": "ondrej-chrastina-tech"
        },
        "source_repository": {
          "type": "modular_content",
          "name": "Source repository",
          "value": [
            "simply007_simply007_github_io"
          ]
        }
      }
    },
    "simply007_kontent_gatsby_benchmark": {
      "system": {
        "id": "16ec0ca3-8525-4154-ba84-0bb7ec712bf6",
        "name": "Simply007/kontent-gatsby-benchmark",
        "codename": "simply007_kontent_gatsby_benchmark",
        "language": "default",
        "type": "repository",
        "collection": "default",
        "sitemap_locations": [],
        "last_modified": "2020-08-14T12:19:40.5439642Z",
        "workflow_step": "published"
      },
      "elements": {
        "name": {
          "type": "text",
          "name": "Name",
          "value": "Simply007/kontent-gatsby-benchmark"
        },
        "summary": {
          "type": "text",
          "name": "Summary",
          "value": "Example site for Kontent as a source for \"Will It Build\". Should/Will be generalized into e.g. a theme."
        },
        "url": {
          "type": "text",
          "name": "URL",
          "value": "https://github.com/Simply007/kontent-gatsby-benchmark"
        },
        "slug": {
          "type": "url_slug",
          "name": "Slug",
          "value": "simply007-kontent-gatsby-benchmark"
        }
      }
    }
  }
}