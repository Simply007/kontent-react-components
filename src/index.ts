export * from './components/rich-text-element';

export { isLinkedItem, isComponent } from './utils/richTextUtils';