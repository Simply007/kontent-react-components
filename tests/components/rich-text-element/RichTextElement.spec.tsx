import React from 'react';
import { render } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import { createDeliveryClient, Elements } from '@kentico/kontent-delivery';
import richTextItem from './richTextItem.json';
import { Element as ParserElement } from "html-react-parser";
import { RichTextElement } from '../../../src';
test('Parse rich text', () => {

  var mockClient = createDeliveryClient({
    projectId: 'dummyClient',
  });

  const response = mockClient.item("dummyItem")
    .map(richTextItem);


  const { debug } = render(<RichTextElement
    richTextElement={response.item.elements["bio"] as Elements.RichTextElement}
    linkedItems={response.linkedItems}
    resolveLinkedItem={undefined}
    resolveImage={undefined}
    resolveLink={undefined}
    resolveDomNode={undefined}
    className="testClassName"
  />);

  debug();
});

describe('<ReactRichTextElement/>', () => {
  it('Empty rich-text value - render properly', () => {
    const testRenderer = TestRenderer.create(
      <RichTextElement richTextElement={{
        value: "<p><br></p>"
      }}
      />,
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('Simple rich-text value resolved simple dom node properly', () => {
    const testRenderer = TestRenderer.create(
      <RichTextElement
        richTextElement={{
          value: "<p>Lorem ipsum with <strong>bold text</strong></p>"
        }}
        resolveDomNode={(domNode, domToReact) => {
          if (domNode instanceof ParserElement) {
            if (domNode.name === "strong") {
              return <b>{domToReact([domNode])}</b>;
            }
          }
        }
        }
      />,
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  // it('Complex rich-text value - render properly no resolution', () => {
  //   const testRenderer = TestRenderer.create(
  //     <ReactRichTextElement richTextElement={sampleComplexValue} />,
  //   );
  //   expect(testRenderer.toJSON()).toMatchSnapshot();
  // });

  // it('Resolve images', () => {
  //   const testRenderer = TestRenderer.create(
  //     <ReactRichTextElement
  //       richTextElement={sampleComplexValue}
  //       images={images}
  //       resolveImage={(image): JSX.Element => (
  //         <img
  //           src={image.url}
  //           alt={image.description ? image.description : image.name}
  //           width="200"
  //         />
  //       )}
  //     />,
  //   );
  //   expect(testRenderer.toJSON()).toMatchSnapshot();
  // });

  // it('Resolve links', () => {
  //   const complexValueRenderer = TestRenderer.create(
  //     <ReactRichTextElement
  //       richTextElement={sampleComplexValue}
  //       links={links}
  //       resolveLink={(link, domNode): JSX.Element => {
  //         return (
  //           // normally a Link component from gatsby package would be used
  //           <a href={`/${link.type}/${link.url_slug}`}>
  //             {domNode.children[0].data}
  //           </a>
  //         );
  //       }}
  //     />,
  //   );
  //   expect(complexValueRenderer.toJSON()).toMatchSnapshot();


  //   const simpleValueRenderer = TestRenderer.create(
  //     <ReactRichTextElement
  //       richTextElement={'<p>This is the page text.</p><ul><li><a data-item-id="1abb6bf1-1e29-4deb-bb0c-b5928ffb0cc9" href="">Test link</a></li></ul>'}
  //       links={[
  //         {
  //           "url_slug": "test-nico",
  //           "type": "page",
  //           "link_id": "1abb6bf1-1e29-4deb-bb0c-b5928ffb0cc9",
  //           "codename": "test_page_nico"
  //         }
  //       ]}
  //       resolveLink={(link, domNode): JSX.Element => {
  //         return (
  //           // normally a Link component from gatsby package would be used
  //           <a href={`/${link.type}/${link.url_slug}`}>
  //             {domNode.children[0].data}
  //           </a>
  //         );
  //       }}
  //     />,
  //   );
  //   expect(simpleValueRenderer.toJSON()).toMatchSnapshot();
  // });

  // it('Resolve linked items', () => {
  //   const testRenderer = TestRenderer.create(
  //     <ReactRichTextElement
  //       richTextElement={sampleComplexValue}
  //       linkedItems={linkedItems}
  //       resolveLinkedItem={(linkedItem): JSX.Element => {
  //         return <pre>{JSON.stringify(linkedItem, undefined, 2)}</pre>;
  //       }}
  //     />,
  //   );
  //   expect(testRenderer.toJSON()).toMatchSnapshot();
  // });

  // it('Resolve geneal node - wrap table element', () => {
  //   const testRenderer = TestRenderer.create(
  //     <ReactRichTextElement
  //       richTextElement={sampleComplexValue}
  //       resolveDomNode={(domNode, domToReact): JSX.Element => {
  //         if (domNode.name === 'table') {
  //           return <div className="table-wrapper">{domToReact([domNode])}</div>;
  //         }
  //       }}
  //     />,
  //   );
  //   expect(testRenderer.toJSON()).toMatchSnapshot();
  // });
});

