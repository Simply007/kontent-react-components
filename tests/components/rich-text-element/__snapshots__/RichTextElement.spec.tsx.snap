// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RichTextElement/> Complex rich-text value - render properly no resolution 1`] = `
<div>
  <p>
    This is Ondřej Chrastina - Developer Advocate with 
    <a
      data-new-window="true"
      href="https://kontent.ai"
      rel="noopener noreferrer"
      target="_blank"
    >
      Kentico Kontent
    </a>
    .
  </p>
  <figure
    data-asset-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
    data-image-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
  >
    <img
      alt=""
      data-asset-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
      data-image-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
      src="https://assets-us-01.kc-usercontent.com:443/0f6785d7-8214-000f-62b3-d9580d692b59/44299668-b37b-4224-a115-1fd66f7d7b36/Yprofile.jpg"
    />
  </figure>
  

  <p>
    He likes to do web sites. This is his latest project:
  </p>
  

  <object
    data-codename="ondrej_chrastina_tech"
    data-rel="link"
    data-type="item"
    type="application/kenticocloud"
  />
  

  <p>
    <br />
    
He also likes OSS. This is his latest repository:
  </p>
  

  <object
    data-codename="simply007_kontent_gatsby_benchmark"
    data-rel="link"
    data-type="item"
    type="application/kenticocloud"
  />
  

  <p>
    <br />
    
On some projects, he was cooperating with 
    <a
      data-item-id="2b805947-7ca5-4e6a-baa5-734a91f3cfa2"
      href=""
    >
      John Doe
    </a>
    .
  </p>
  

  <p>
    You could take a look at their 
    <a
      href="https://google.com"
      title="sample link"
    >
      latest project
    </a>
    .
  </p>
  

  <p>
    Check out also this interesting repository
  </p>
  

  <object
    data-codename="n5b76e971_4709_01f2_e8ea_36a9716d7b42"
    data-rel="component"
    data-type="item"
    type="application/kenticocloud"
  />
</div>
`;

exports[`<RichTextElement/> Empty rich-text value - render properly 1`] = `
<div>
  <p>
    <br />
  </p>
</div>
`;

exports[`<RichTextElement/> Resolve geneal node - wrap table element 1`] = `
<div>
  <p>
    This is Ondřej Chrastina - Developer Advocate with 
    <a
      data-new-window="true"
      href="https://kontent.ai"
      rel="noopener noreferrer"
      target="_blank"
    >
      Kentico Kontent
    </a>
    .
  </p>
  <figure
    data-asset-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
    data-image-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
  >
    <img
      alt=""
      data-asset-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
      data-image-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
      src="https://assets-us-01.kc-usercontent.com:443/0f6785d7-8214-000f-62b3-d9580d692b59/44299668-b37b-4224-a115-1fd66f7d7b36/Yprofile.jpg"
    />
  </figure>
  

  <p>
    He likes to do web sites. This is his latest project:
  </p>
  

  <object
    data-codename="ondrej_chrastina_tech"
    data-rel="link"
    data-type="item"
    type="application/kenticocloud"
  />
  

  <p>
    <br />
    
He also likes OSS. This is his latest repository:
  </p>
  

  <object
    data-codename="simply007_kontent_gatsby_benchmark"
    data-rel="link"
    data-type="item"
    type="application/kenticocloud"
  />
  

  <p>
    <br />
    
On some projects, he was cooperating with 
    <a
      data-item-id="2b805947-7ca5-4e6a-baa5-734a91f3cfa2"
      href=""
    >
      John Doe
    </a>
    .
  </p>
  

  <p>
    You could take a look at their 
    <a
      href="https://google.com"
      title="sample link"
    >
      latest project
    </a>
    .
  </p>
  

  <p>
    Check out also this interesting repository
  </p>
  

  <object
    data-codename="n5b76e971_4709_01f2_e8ea_36a9716d7b42"
    data-rel="component"
    data-type="item"
    type="application/kenticocloud"
  />
</div>
`;

exports[`<RichTextElement/> Resolve images 1`] = `
<div>
  <p>
    This is Ondřej Chrastina - Developer Advocate with 
    <a
      data-new-window="true"
      href="https://kontent.ai"
      rel="noopener noreferrer"
      target="_blank"
    >
      Kentico Kontent
    </a>
    .
  </p>
  <img
    alt="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
    src="https://assets-us-01.kc-usercontent.com:443/0f6785d7-8214-000f-62b3-d9580d692b59/44299668-b37b-4224-a115-1fd66f7d7b36/Yprofile.jpg"
    width="200"
  />
  

  <p>
    He likes to do web sites. This is his latest project:
  </p>
  

  <object
    data-codename="ondrej_chrastina_tech"
    data-rel="link"
    data-type="item"
    type="application/kenticocloud"
  />
  

  <p>
    <br />
    
He also likes OSS. This is his latest repository:
  </p>
  

  <object
    data-codename="simply007_kontent_gatsby_benchmark"
    data-rel="link"
    data-type="item"
    type="application/kenticocloud"
  />
  

  <p>
    <br />
    
On some projects, he was cooperating with 
    <a
      data-item-id="2b805947-7ca5-4e6a-baa5-734a91f3cfa2"
      href=""
    >
      John Doe
    </a>
    .
  </p>
  

  <p>
    You could take a look at their 
    <a
      href="https://google.com"
      title="sample link"
    >
      latest project
    </a>
    .
  </p>
  

  <p>
    Check out also this interesting repository
  </p>
  

  <object
    data-codename="n5b76e971_4709_01f2_e8ea_36a9716d7b42"
    data-rel="component"
    data-type="item"
    type="application/kenticocloud"
  />
</div>
`;

exports[`<RichTextElement/> Resolve linked items from richText element with richText containing multiple level of linked items does not resolve lower level without linkedItems attribute 1`] = `
<div>
  <div
    className="row"
  >
    <div>
      <div
        className="column"
      >
        <div>
          <p>
            First column
          </p>
        </div>
      </div>
      <div
        className="column"
      >
        <div>
          <p>
            Second column
          </p>
        </div>
      </div>
      

      <div
        className="column"
      >
        <div>
          <p>
            Third column
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<RichTextElement/> Resolve linked items from richText element with richText containing single level of linked items 1`] = `
<div>
  <p>
    This is Ondřej Chrastina - Developer Advocate with 
    <a
      data-new-window="true"
      href="https://kontent.ai"
      rel="noopener noreferrer"
      target="_blank"
    >
      Kentico Kontent
    </a>
    .
  </p>
  <figure
    data-asset-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
    data-image-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
  >
    <img
      alt=""
      data-asset-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
      data-image-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
      src="https://assets-us-01.kc-usercontent.com:443/0f6785d7-8214-000f-62b3-d9580d692b59/44299668-b37b-4224-a115-1fd66f7d7b36/Yprofile.jpg"
    />
  </figure>
  

  <p>
    He likes to do web sites. This is his latest project:
  </p>
  

  <h1>
    Linked item
  </h1>
  <pre>
    {
  "elements": {
    "name": {
      "name": "Name",
      "type": "text",
      "value": "ondrej.chrastina.tech"
    },
    "summary": {
      "name": "Summary",
      "type": "text",
      "value": "Ondřej Chrastina's personal site"
    },
    "url": {
      "name": "URL",
      "type": "text",
      "value": "https://ondrej.chrastina.tech"
    },
    "slug": {
      "name": "Slug",
      "type": "url_slug",
      "value": "ondrej-chrastina-tech"
    },
    "source_repository": {
      "name": "Source repository",
      "type": "modular_content",
      "value": [
        "simply007_simply007_github_io"
      ],
      "linkedItems": []
    }
  },
  "system": {
    "codename": "ondrej_chrastina_tech",
    "collection": "default",
    "id": "884bd3cf-98b6-438e-b50d-a593a3bad623",
    "language": "default",
    "lastModified": "2020-08-14T13:18:13.858424Z",
    "name": "ondrej.chrastina.tech",
    "sitemapLocations": [],
    "type": "website",
    "workflowStep": "published"
  }
}
  </pre>
  ;
  

  <p>
    <br />
    
He also likes OSS. This is his latest repository:
  </p>
  

  <h1>
    Linked item
  </h1>
  <pre>
    {
  "elements": {
    "name": {
      "name": "Name",
      "type": "text",
      "value": "Simply007/kontent-gatsby-benchmark"
    },
    "summary": {
      "name": "Summary",
      "type": "text",
      "value": "Example site for Kontent as a source for \\"Will It Build\\". Should/Will be generalized into e.g. a theme."
    },
    "url": {
      "name": "URL",
      "type": "text",
      "value": "https://github.com/Simply007/kontent-gatsby-benchmark"
    },
    "slug": {
      "name": "Slug",
      "type": "url_slug",
      "value": "simply007-kontent-gatsby-benchmark"
    }
  },
  "system": {
    "codename": "simply007_kontent_gatsby_benchmark",
    "collection": "default",
    "id": "16ec0ca3-8525-4154-ba84-0bb7ec712bf6",
    "language": "default",
    "lastModified": "2020-08-14T12:19:40.5439642Z",
    "name": "Simply007/kontent-gatsby-benchmark",
    "sitemapLocations": [],
    "type": "repository",
    "workflowStep": "published"
  }
}
  </pre>
  ;
  

  <p>
    <br />
    
On some projects, he was cooperating with 
    <a
      data-item-id="2b805947-7ca5-4e6a-baa5-734a91f3cfa2"
      href=""
    >
      John Doe
    </a>
    .
  </p>
  

  <p>
    You could take a look at their 
    <a
      href="https://google.com"
      title="sample link"
    >
      latest project
    </a>
    .
  </p>
  

  <p>
    Check out also this interesting repository
  </p>
  

  <h1>
    Component
  </h1>
  <pre>
    {
  "elements": {
    "name": {
      "name": "Name",
      "type": "text",
      "value": "Gatsby packages repository"
    },
    "slug": {
      "name": "Slug",
      "type": "url_slug",
      "value": ""
    },
    "summary": {
      "name": "Summary",
      "type": "text",
      "value": "Gatsby packages repository containing the source plugin, components package, and a bunch of Gatsby examples."
    },
    "url": {
      "name": "URL",
      "type": "text",
      "value": "https://github.com/Kentico/kontent-gatsby-packages"
    }
  },
  "system": {
    "codename": "n5b76e971_4709_01f2_e8ea_36a9716d7b42",
    "collection": "default",
    "id": "5b76e971-4709-01f2-e8ea-36a9716d7b42",
    "language": "default",
    "lastModified": "2022-02-14T09:33:27.4528882Z",
    "name": "5b76e971-4709-01f2-e8ea-36a9716d7b42",
    "sitemapLocations": [],
    "type": "repository",
    "workflowStep": null
  }
}
  </pre>
  ;
</div>
`;

exports[`<RichTextElement/> Resolve links 1`] = `
<div>
  <p>
    This is Ondřej Chrastina - Developer Advocate with 
    <a
      data-new-window="true"
      href="https://kontent.ai"
      rel="noopener noreferrer"
      target="_blank"
    >
      Kentico Kontent
    </a>
    .
  </p>
  <figure
    data-asset-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
    data-image-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
  >
    <img
      alt=""
      data-asset-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
      data-image-id="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
      src="https://assets-us-01.kc-usercontent.com:443/0f6785d7-8214-000f-62b3-d9580d692b59/44299668-b37b-4224-a115-1fd66f7d7b36/Yprofile.jpg"
    />
  </figure>
  

  <p>
    He likes to do web sites. This is his latest project:
  </p>
  

  <object
    data-codename="ondrej_chrastina_tech"
    data-rel="link"
    data-type="item"
    type="application/kenticocloud"
  />
  

  <p>
    <br />
    
He also likes OSS. This is his latest repository:
  </p>
  

  <object
    data-codename="simply007_kontent_gatsby_benchmark"
    data-rel="link"
    data-type="item"
    type="application/kenticocloud"
  />
  

  <p>
    <br />
    
On some projects, he was cooperating with 
    <a
      href="/person/john-doe"
    >
      John Doe
    </a>
    .
  </p>
  

  <p>
    You could take a look at their 
    <a
      href="https://google.com"
      title="sample link"
    >
      latest project
    </a>
    .
  </p>
  

  <p>
    Check out also this interesting repository
  </p>
  

  <object
    data-codename="n5b76e971_4709_01f2_e8ea_36a9716d7b42"
    data-rel="component"
    data-type="item"
    type="application/kenticocloud"
  />
</div>
`;

exports[`<RichTextElement/> Resolve links 2`] = `
<div>
  <p>
    This is the page text.
  </p>
  <ul>
    <li>
      <a
        href="/page/test-nico"
      >
        Test link
      </a>
    </li>
  </ul>
</div>
`;

exports[`<RichTextElement/> Simple rich-text value resolved simple dom node properly 1`] = `
<div>
  <p>
    Lorem ipsum with 
    <b>
      bold text
    </b>
  </p>
</div>
`;
